### ISSUE TITLE
Variable having type type cannot act as base class?

### ISSUE THREAD

**Issue by NeilGirdhar** [microsoft/pyright#4690]
```python
def f(x: type) -> object:
    class Y(x):  #  error: Expected type expression but received "type"
        pass
    return Y()

f(int)
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.296, which I just published. It will also be included in a future release of pylance.
